app/assets/javascripts/controllers/generic_object/generic_object_definition_controller.js

```diff
@@ -0,0 +1,75 @@
+ManageIQ.angular.app.controller('genericObjectDefinitionFormController', ['$http', '$scope', 'miqService', 'genericObjectSubscriptionService', function($http, $scope, miqService, genericObjectSubscriptionService) {
+  var init = function() {
+    hideAndClearForm();
+
+    $scope.newRecord = true;
+    $scope.showSingleItem = false;
+
+    genericObjectSubscriptionService.subscribeToShowAddForm(showAddForm);
+    genericObjectSubscriptionService.subscribeToTreeClicks(showSelectedItem);
+    genericObjectSubscriptionService.subscribeToRootTreeclicks(showAllItems);
+  };
+
+  var showAddForm = function(_response) {
+    $scope.clearForm();
+    $scope.showAddForm = true;
+    sendDataWithRx({eventType: 'deselectTreeNodes'});
+  };
+
+  var showAllItems = function(response) {
+    $scope.genericObjectList = response;
+    $scope.showSingleItem = false;
+  };
+
+  var addedGenericObject = function(data) {
```

Comment:
Ah, ok, that's the part I didn't see in the first place. Ok, so, not sure why I didn't use `$http` here, but I think I'll just do that instead. It'll be more consistent, and it'll work as intended, yes?